aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -76,99 +139,533 @@ void filterEngineConfigs(
 
 struct CacheKey {
   ConvolutionParams params;
-  uint8_t input_alignment;
-  uint8_t weight_alignment;
-  uint8_t output_alignment;
+  cudnnBackendDescriptorType_t operation;
+  uint8_t x_alignment;
+  uint8_t w_alignment;
+  uint8_t y_alignment;
 };
 
-// FIXME: make this thread-safe by reusing the benchmark cache in Conv_v7.cpp
-std::unordered_map<CacheKey, cudnn_frontend::ManagedOpaqueDescriptor, ParamsHash<CacheKey>, ParamsEqual<CacheKey>> engine_cache;
+struct CacheKeyFused {
+  ConvolutionParams params;
+  // No op here because it is assumed to be a forward conv op
+  uint8_t x_alignment;
+  uint8_t w_alignment;
+  uint8_t y_alignment;
+  uint8_t z_alignment;
+  uint8_t b_alignment;
+  // TODO: does it make sense to have this in the key? but alpha is a graph-level param...
+  float alpha;
+};
 
-}
+template <typename T, typename KeyType>
+struct BenchmarkCache {
+std::mutex mutex;
+std::unordered_map<KeyType, cudnn_frontend::ExecutionPlan, ParamsHash<KeyType>, ParamsEqual<KeyType>> engine_cache;
 
-void raw_cudnn_convolution_forward_out(
-    const Tensor& output, const Tensor& input, const Tensor& weight,
-    IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups,
-    bool benchmark, bool deterministic, bool allow_tf32)
-{
-  TORCH_CHECK(!benchmark, "not supported yet");
-  if (output.numel() == 0) {
-    return;
+// TODO: is this thread safe if cache is updated? is pointer stale?
+cudnn_frontend::ExecutionPlan* find(const KeyType& key) {
+  std::lock_guard<std::mutex> guard(mutex);
+  auto it = engine_cache.find(key);
+  if (it == engine_cache.end()) {
+    return NULL;
   }
+  // TODO: probably want ExecutionPlan copy constructor or better way to return
+  return &(it->second);
```

Comment:
References are typically not invalidated (unless the element in question is erased) so this is ok-ish, but if you decide to use faster container than unordered_map that may no longer hold.